shims/spark310/src/main/scala/com/nvidia/spark/rapids/shims/spark310/ParquetCachedBatchSerializer.scala

```diff
@@ -681,9 +761,72 @@ class ParquetCachedBatchSerializer extends CachedBatchSerializer with Arm {
      * and return the CachedBatch when next is called.
      */
     private class InternalRowToCachedBatchIterator extends Iterator[CachedBatch]() {
+      // is there a type that spark doesn't support by default in the schema?
+      val hasUnsupportedType = cachedAttributes.toStructType.fields.exists { field =>
```

Comment:
Why convert it to a struct?  Why not just 

```
cachedAttributes.exists( attr => !isTypeSupportedByParquet(attr.dataType))
```

Reply:
nit: I Still see this in the code on line 535

Reply:
Sorry, I didn't see there was another variable named `cacheAttributes` instead of `cachedAttributes`. It's done now. PTAL